integration-libs/opf/base/root/model/opf.model.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ViewContainerRef } from '@angular/core';
+
+export interface OpfPaymentMetadata {
+  termsAndConditionsChecked: boolean;
+  selectedPaymentOptionId: number | undefined;
+  defaultSelectedPaymentOptionId?: number;
+  isPaymentInProgress: boolean;
+  paymentSessionId: string | undefined;
+}
+
+export interface KeyValuePair {
+  key: string;
+  value: string;
+}
+
+export type MerchantCallback = (
+  response?: SubmitResponse | SubmitCompleteResponse
+) => void | Promise<void>;
+
+export interface GlobalOpfPaymentMethods {
+  getRedirectParams?(): Array<KeyValuePair>;
+  submit?(options: {
+    cartId: string;
+    additionalData: Array<KeyValuePair>;
+    submitSuccess: MerchantCallback;
+    submitPending: MerchantCallback;
+    submitFailure: MerchantCallback;
+    paymentMethod: PaymentMethod;
+  }): Promise<boolean>;
+  submitComplete?(options: {
+    cartId: string;
+    additionalData: Array<KeyValuePair>;
+    submitSuccess: MerchantCallback;
+    submitPending: MerchantCallback;
+    submitFailure: MerchantCallback;
+  }): Promise<boolean>;
+  submitCompleteRedirect?(options: {
+    cartId: string;
+    additionalData: Array<KeyValuePair>;
+    submitSuccess: MerchantCallback;
+    submitPending: MerchantCallback;
+    submitFailure: MerchantCallback;
+  }): Promise<boolean>;
+  throwPaymentError?(errorOptions?: ErrorDialogOptions): void;
+  startLoadIndicator?(): void;
+  stopLoadIndicator?(): void;
+}
+
+export interface PaymentBrowserInfo {
+  acceptHeader?: string;
+  colorDepth?: number;
+  javaEnabled?: boolean;
+  javaScriptEnabled?: boolean;
+  language?: string;
+  screenHeight?: number;
+  screenWidth?: number;
+  userAgent?: string;
+  timeZoneOffset?: number;
+  ipAddress?: string;
+  originUrl?: string;
+}
+
+export interface SubmitRequest {
+  browserInfo?: PaymentBrowserInfo;
+  paymentMethod?: string;
+  encryptedToken?: string;
+  cartId?: string;
+  channel?: string;
+  additionalData?: Array<KeyValuePair>;
+}
+
+export interface SubmitInput {
+  additionalData: Array<KeyValuePair>;
+  paymentSessionId: string;
+  cartId: string;
+  callbackArray: [MerchantCallback, MerchantCallback, MerchantCallback];
```

Comment:
[MEDIUM]

The property `callbackArray` is enigmatic because there are three anonymous empty functions passed, making it hard to reason about. You can only learn the actual meanings of those three items in the array from the source code of the `OpfPaymentHostedFieldsService`, specifically the `paymentResponseHandler()` function. Therefore, let's refactor this property from `callbacksArray: [MerchantCallback,MerchantCallback,MerchantCallback]` to an object with semantic keys:

```ts
callbacks: {
  submitSuccess: MerchantCallback, 
  submitPending: MerchantCallback,
  submitFailure: MerchantCallback,
}
```

Reply:
[OUTDATED] Done.